### PR TITLE
feat(IC-1579): add the Apalache infrastructure to Bazel/Docker

### PR DESCRIPTION
**Pull request by oggy-dfin**
This is a preliminary PR to do add the infra (JRE and downloading Apalache) necessary to run Apalache, which we'll use to test the TLA-code link.

### CODE REVIEW

File: gitlab-ci/container/Dockerfile [hunk 1]
```diff
@@ -84,6 +84,10 @@ RUN . /opt/nvm/nvm.sh && \
     nvm install ${dependency_mgmt_nns_dapp_node_version} && \
     nvm install ${dependency_mgmt_default_node_version}
 
+# Install JRE for TLA/Apalache checks
+RUN apt -yq update && \
```
Comment (marko-k0) [inline gitlab-ci/container/Dockerfile:88]: Introducing a new layer just to add a new package is too much. I think you can add this package to `files/packages.common`.

Comment (oggy-dfin) [reply gitlab-ci/container/Dockerfile]: Thanks - moved it to packages.common now

**Review by nmattia — COMMENTED** [review]
@oggy-dfin actually how about adding it once the TLA tests are ready? That way we'll have a better idea of what you need this for and can better review this

**Comment by oggy-dfin** [comment]
@nmattia there's a draft PR here:
https://github.com/dfinity/ic/pull/913
But it's big so I figure I'd help out both the IDX and the NNS folks with their reviews by doing it in two steps. It also helps me not chase around the latest Docker tags until the PR is merged.

**Review by mbjorkqvist — APPROVED** [review]
If updating the image is fine for all the other workflows, I assume it is ok for the rosetta release also.

**Comment by oggy-dfin** [comment]
Closing in favor of https://github.com/dfinity/ic/pull/1307

**Pull request closed without merging**